src/knx/memory.cpp

```diff
@@ -84,29 +91,37 @@ void Memory::readMemory()
 
 void Memory::writeMemory()
 {
-    uint8_t* buffer = _data;
-    buffer = pushWord(_deviceObject.manufacturerId(), buffer);
-    buffer = pushByteArray(_deviceObject.hardwareType(), LEN_HARDWARE_TYPE, buffer);
-    buffer = pushWord(_deviceObject.version(), buffer);
+    // first get the necessary size of the writeBuffer
+    size_t writeBufferSize = 16;
```

Comment:
I have no idea.
The change was made by you before I started to work on the flash branch ;)
https://github.com/thelsing/knx/commit/4b873aebd60ec14521826453e2c3ba74726fcdb9

Reply:
I think it should be alignToPageSize(_metadataSize) as in readMemory.  Not sure how 16 came to be.

Reply:
changed to writeBufferSize = _metadataSize.
No need for alignToPageSize here because the writebuffer must only hold the real data, not the "spacing" data.